### PR TITLE
chore: update xchain reference blocks

### PR DESCRIPTION
**Pull request by maharifu**
Respond to new request type with updated reference blocks for all supported chains.

# Related Github tickets

- https://github.com/VolumeFi/paloma/issues/1637

# Background

At startup, pigeon validates that it's talking to the right chain by using the reference block height and hash. These values are introduced when first integrating a chain. However, as time passes, some RPC providers will prune block information, which makes it impossible for pigeon to verify the blocks.

To solve this, we introduce a new periodic task that at each 10000 blocks, sends a message requesting pigeons to update the reference block information (height and hash) for each supported chain.

We add a new queue - `evm/<CHAIN>/reference-block` - and follow the same flow from ValidatorBalances.

Needs the message definitions from https://github.com/palomachain/paloma/pull/1175

# Testing completed

- [x] test coverage exists or has been added/updated
- [x] tested in a private testnet

# Breaking changes

- [x] I have checked my code for breaking changes

### CODE REVIEW

**Comment by byte-bandit** [comment]
You'll need to bump the Paloma dep in order to pull in the proto definitions.

**Comment by byte-bandit** [comment]
FYI you can also change the dependency against a revision in order to pull in the changes if you want to move forward with this PR, but I'll ask you to submit another PR with the proper dependency bump before we release.

**Comment by maharifu** [comment]
> FYI you can also change the dependency against a revision in order to pull in the changes if you want to move forward with this PR, but I'll ask you to submit another PR with the proper dependency bump before we release.

I wanted to avoid two PRs because of this, but I'll change this anyway so we can move, then we merge the one in paloma and I change this back before merging.

Anyway, when doing this, the tests fail. [This commit](https://github.com/palomachain/paloma/commit/0df4a10502795e09809ed4a284abb3b4930ff603) changes the signature of `GetAllQueueNames` mock, but it doesn't seem to align with the implementation. Is this correct?

**Comment by byte-bandit** [comment]
Yeah, there are some outdated mocks in Paloma. This PR should address it: https://github.com/palomachain/paloma/pull/1178

**Comment by maharifu** [comment]
Updated the PR to point paloma dependency to latest commit after merging https://github.com/palomachain/paloma/pull/1175.

File: chain/evm/compass.go [hunk 1]
```diff
@@ -608,6 +608,47 @@ func (t compass) provideEvidenceForValidatorBalance(ctx context.Context, queueTy
...
 }
 
+func (t compass) provideEvidenceForReferenceBlock(ctx context.Context, queueTypeName string, msgs []chain.MessageWithSignatures) error {
+	logger := liblog.WithContext(ctx).WithField("queue-type-name", queueTypeName)
+	logger.Debug("start processing reference block request")
+
+	var g whoops.Group
+	for _, msg := range msgs {
+		g.Add(
+			whoops.Try(func() {
```
Comment (byte-bandit) [inline chain/evm/compass.go:618]: Remove the `whoops` lib usages.

**Review by byte-bandit — CHANGES_REQUESTED** [review]
Good stuff, just one small thing as we try to phase out `whoops`. Did you test this successfully in a private network?

File: chain/evm/compass.go (hunk 1 above)
Comment (maharifu) [reply chain/evm/compass.go]: I removed it. We now return an error after failing a single message, which should be what we wanted anyway.

**Comment by maharifu** [comment]
> Good stuff, just one small thing as we try to phase out `whoops`. Did you test this successfully in a private network?

Yes, I did test it in a private testnet, with a snapshot from mainnet. I could see the reference blocks being updated in chains-infos.

**Review by byte-bandit — APPROVED** [review]
LGTM!

**Pull request merged**